app/src/features/dataset_explorer/DatasetPreview.tsx

```diff
@@ -3,8 +3,16 @@ import DataTable from "./DataTable";
 import useDatasetStore from "../../utils/useDatasetStore";
 import { Dataset } from "../../utils/DatasetTypes";
 
-function getTableViewColumns(dataset: Dataset) {
-  return dataset.metadata.fields.map((field) => ({
+/* NOTE: This file is currently unused */
+
+function getTableViewColumns(
+  dataset: Dataset
+): { Header: string; accessor: string }[] {
+  const fields =
```

Comment:
nit: I prefer `const fields = dataset.metadata.fields || []`. I know falsyness can be a divisive issue, but for complex data types I think it works well since it makes sure you capture both null and undefined, and it's more concise.

Reply:
Your download changes actually make this required again so I'm removing this but ack :)